tombstone_gc.hh

```diff
@@ -113,5 +117,31 @@ public:
     future<> flush_pending_repair_time_update(replica::database& db);
 };
 
+class tombstone_gc_before_getter {
+public:
+    using get_gc_before_for_range_result = tombstone_gc_state::get_gc_before_for_range_result;
+
+private:
+    tombstone_gc_state _tombstone_gc;
+
+public:
+    tombstone_gc_before_getter() : _tombstone_gc(nullptr) { }
```

Comment:
Do we need the default constructor? Seems like a mine.

Reply:
We did need it for the transition, I'll check whether I can drop it at the end.

Reply:
Note that with default-constructed `tombstone_gc_before_getter`, tombstones can't be garbage collected, so it is safe in that regard.

Reply:
Added named consrtuctors and removed the default one in v4.